impress-org/givewp
# Refactor: Update Email template settings in the form builder

## Description

Reference: slack conversation about the cancel option: https://lw.slack.com/archives/C04SLRDD9CK/p1692899169132059

~~This PR includes several changes revolving around the Email Notification Settings. The major change involves the option that allows users to either set the email template fields or cancel the field values~~. 

~~This was done including by including  local state within the email template component to capture all the field changes {```emailTemplateFieldValues``` } and dispatching the actual Form Setting changes when the Set button is clicked.  The original option config is used to update the local state when the Cancel button is clicked to revert the field values~~.

~~Both action buttons "Set and close" & "Cancel" were moved down from the parent container to the EmailTemplateSettings component. I Initially I had looked into raising the state up to the parent in order the Submit the changes. Doing so required moving most the config logic from within the EmailTemplateSettings to the parent also. This made it harder to follow and maintain. Because of this I opted to move the action buttons down and absolutely position them to achieve an identical layout~~. 

The changes reflect the QA tasks below, along with style, typescript type & text changes.

QA Tasks completed within PR: 
- [Email preview has fatal error](https://app.asana.com/0/1203547864809090/1205076251265810)
- [Email Template: Change "save and close" button on modal to "close"](https://app.asana.com/0/1203547864809090/1204927308757485)
- [ Increase spacing between radio buttons under Notification](https://app.asana.com/0/1203547864809090/1204927308757439)
- [ Populate email address field](https://app.asana.com/0/1203547864809090/1204927308757453)


## Visuals

<img width="1720" alt="Screen Shot 2023-08-24 at 1 00 20 PM" src="https://github.com/impress-org/givewp/assets/75056371/bfa716b8-7221-44ba-9cb2-31f0a7e604c2">
<img width="1720" alt="Screen Shot 2023-08-24 at 1 00 13 PM" src="https://github.com/impress-org/givewp/assets/75056371/1717229a-68cd-44a9-a284-9a5b78b8bc83">


## Testing Instructions 
Please note this PR no longer contains logic to cancel the selected email options. As the logic was included than removed in favor of a possible undo feature with gutenberg. 

Please test that the email fields are working as they previously did.
Verify no fatal errors occur on email preview.
Verify default email address is populated in the send test email input.

https://www.loom.com/share/580ee957a673491ea21e01a127ed74f4?sid=93eed98b-fc59-4ed3-836f-aa7064892085

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205360468064600
